### PR TITLE
Add batch and overwrite options to historical backfill

### PR DESCRIPTION
**Pull request by acmiyaguchi**
This makes it possible to batch import executive report data. This has the possibility of overwriting the entire repository, so a confirmation is needed to continue.

This has been used to import both weekly and monthly data to `net-mozaws-prod-us-west-2-pipeline-analysis/amiyaguchi/topline_summary`.

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/python_mozetl/26)

### CODE REVIEW

File: mozetl/topline/historical_backfill.py [hunk 1]
```diff
@@ -57,16 +56,42 @@ def backfill_topline_summary(historical_df, path):
...
+    write_mode = "overwrite" if overwrite else "error"
+    writer = df.write.mode(write_mode)
+
     # Use the same parititoning scheme as topline_summary
-    df.write.partitionBy('report_start').parquet(path)
+    if batch:
+        writer = writer.partitionBy('report_start')
+    else:
+        # The csv file should be read from
+        # telemetry-private-analysis=2/executive-report. This folder contains csv
```
Comment (mreid-moz) [inline mozetl/topline/historical_backfill.py]: should be `telemetry-private-analysis-2/executive-report-<mode>/data/`

File: mozetl/topline/historical_backfill.py [hunk 2]
```diff
@@ -36,13 +36,12 @@ def format_output_path(bucket, prefix):
     return "s3://{}/{}".format(bucket, prefix)
 
 
-def backfill_topline_summary(historical_df, path):
+def backfill_topline_summary(historical_df, path, batch=False, overwrite=False):
     """ Backfill the topline summary with the historical dataframe.
 
     :historical_df dataframe: Data from the v4 historical data csv
     :path str: spark compatible path string to save partitioned parquet data
```
Comment (mreid-moz) [inline mozetl/topline/historical_backfill.py]: Please add new arguments to the docstring too

File: mozetl/topline/historical_backfill.py [hunk 3]
```diff
@@ -57,16 +56,42 @@ def backfill_topline_summary(historical_df, path):
...
+@click.option('--batch/--no-batch', default=False,
+        help='Used when the csv file contains multiple reporting periods')
+@click.option('--overwrite/--no-overwrite', default=False,
+        help='Overwrite existing data. Caution when applying this batch mode,'
+             'since it will overwrite all data in the repository.')
+def main(source_s3_path, mode, bucket, prefix, batch, overwrite):
+
+    if batch and overwrite:
+        click.confirm('CAUTION: Using --batch and --overwrite will completely '
+                'overwrite the prefix with input data. Do you want to '
```
Comment (mreid-moz) [inline mozetl/topline/historical_backfill.py]: Can you make it show the actual S3 path that will be wiped out / overwritten?

File: mozetl/topline/historical_backfill.py [hunk 4]
```diff
@@ -57,16 +56,42 @@ def backfill_topline_summary(historical_df, path):
...
 @click.argument('source_s3_path')
 @click.argument('mode', type=click.Choice(['weekly', 'monthly']))
 @click.argument('bucket')
 @click.option('--prefix', default='topline_summary/v1')
-def main(source_s3_path, mode, bucket, prefix):
+@click.option('--batch/--no-batch', default=False,
+        help='Used when the csv file contains multiple reporting periods')
+@click.option('--overwrite/--no-overwrite', default=False,
+        help='Overwrite existing data. Caution when applying this batch mode,'
+             'since it will overwrite all data in the repository.')
```
Comment (mreid-moz) [inline mozetl/topline/historical_backfill.py]: The word `repository` here might be confusing. Maybe use something like `in the specified S3 path`.

**Comment by acmiyaguchi** [comment]
@mreid-moz RFAL

**Pull request merged**